### ISSUE TITLE
How to set the timeout time in useFetch

### ISSUE THREAD

**Issue by luckilyhang** [nuxt/nuxt#20433]
### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.13.0`
- Nuxt Version:     `3.3.1`
- Nitro Version:    `1.0.0`
- Package Manager:  `yarn@1.22.17`
- Builder:          `vite`
- User Config:      `srcDir`, `modules`, `elementPlus`, `app`, `hooks`, `nitro`, `plugins`, `css`, `experimental`, `render`, `sourcemap`, `vue`, `modern`
- Runtime Modules:  `@pinia/nuxt@0.4.7`, `@element-plus/nuxt@1.0.4`
- Build Modules:    `

### Reproduction

Unable to set request time in useFetch request function

### Describe the bug

Unable to set request time in useFetch request function

**Comment by danielroe** [comment]
Would you explain a little more about what you want and the motivations behind it?

**Comment by luckilyhang** [comment]
> Would you explain a little more about what you want and the motivations behind it?



> I want to handle request timeout, such as axios. default. timeout: 1000 * 10

**Comment by luckilyhang** [comment]
> Would you explain a little more about what you want and the motivations behind it?
I want to handle request timeout, such as axios. default. timeout: 1000 * 10

**Comment by pi0** [comment]
Linked upstream issue: https://github.com/unjs/ofetch/issues/21

I think this would be a nice feature to the fetch 👍🏼

**Comment by luckilyhang** [comment]
> Linked upstream issue: [unjs/ofetch#21](https://github.com/unjs/ofetch/issues/21)
> 
> I think this would be a nice feature to the fetch 👍🏼
Can I define options. timeout=1000 * 10 in the options of useFetch?  

be similar to    options.baseURL    options.params  ...

**Comment by danielroe** [comment]
As soon as this is implemented in `ofetch` it will be useable within Nuxt, so I think we can close this issue.

**Comment by martinschilliger** [comment]
Timeout is now supported: https://github.com/unjs/ofetch#%EF%B8%8F-timeout

But if I add it as an option in `useFetch` it has no effect. Is there something that needs to be done before this will work?
```
await useFetch("/books", {
  timeout: 5000, // Timeout after 5 seconds
  onResponse({ request, response, options }) {
    console.log({ response });
  }
}).catch((err) => {
  console.error(err);
});
```

**Comment by danielroe** [comment]
If it isn't working for you, please do open a new issue with a reproduction. 🙏

**Comment by danielroe** [comment]
What makes you think it is not working?

**Comment by martinschilliger** [comment]
If I activate throttling in Edge it still never fails in my local project. I will try to create a reproduction somewhere.

**Comment by martinschilliger** [comment]
I think I created a dome where the same problem exists:
https://stackblitz.com/edit/nuxt-starter-3foten?file=app.vue

**Comment by carloeusebi** [comment]
I can second this, the timeout has no effect for me too.

**Comment by manniL** [comment]
Let's follow up in https://github.com/nuxt/nuxt/issues/24354
